proposals/0345-android-12-issues.md

```diff
@@ -0,0 +1,322 @@
+
+## Motivation
+
+Android 12 adds multiple restrictions on how apps can work in the background and access device resources. Some of the new restrictions that affect how SDL Android works are:
+
+- Apps will no longer be able to start foreground services from the background except for in a few select cases. This will directly impact how apps are able to start their own `SdlService` implementation.
+
+- Android 12 introduces new runtime Bluetooth permissions that will be required by the library to successfully establish a Bluetooth connection with the head unit ([`BLUETOOTH_CONNECT` and `BLUETOOTH_SCAN`](https://developer.android.com/about/versions/12/features#bluetooth-permissions)).
+ 
+- Apps will need to explicitly set the exported flag for any services, receivers, and activities that have an `intent-filter` in the manifest.
+
+- `PendingIntents` now require the mutability flag to be set in all cases which impacts some notifications sent by the router service.
+
+- Android 12 may sometimes choose to hide notifications from services for up to 10 seconds to allow short-running services to finish without interrupting the user. This is in most cases a good change. However, in some cases in the library, the notifications should be updated to be displayed immediately.
+
+## Proposed solution
+
+### Foreground Services
+With the new restrictions around starting foreground services from the background we will need to meet one of the exceptions that Android 12 would allow to start a foreground service from the background or we would need to start the foreground service from a foreground context.
+
+PendingIntents allows us to send an intent on behalf of another app using the context and permissions of the application that created the PendingIntent. Since the `SdlRouterService` will run in the foreground, the `SdlRouterService` could create a PendingIntent to start an apps  `SdlService`. Since the `SdlRouterService` is creating the PendingIntent, the intent would be sent from the `SdlRouterServices` Context.
+
+The PendingIntent in the `SdlRouterService` can be given to the `SdlBroadcastReceiver`. At this point the `SdlReceiver` will be able to send the PendingIntent with and updated Intent where the developer specifies their unique `SdlService` class. By calling `PendingIntent.send()` we are starting a new foreground service of the unique `SdlService` class from the context of the `SdlRouterService`. Since the `SdlRouterService` is running in the foreground we are not longer trying to start a foreground service from the background.
```

Comment:
```suggestion
The `PendingIntent` in the `SdlRouterService` can be given to the `SdlBroadcastReceiver`. At this point the `SdlReceiver` will be able to send the `PendingIntent` with and updated Intent where the developer specifies their unique `SdlService` class. By calling `PendingIntent.send()` we are starting a new foreground service of the unique `SdlService` class from the context of the `SdlRouterService`. Since the `SdlRouterService` is running in the foreground we are no longer trying to start a foreground service from the background.
```